### PR TITLE
fixed split axis logic for mean/var, both also now use array as well

### PR DESCRIPTION
**Pull request by coquelin77**
## Description

Fixed bug

Fixes: #421 

Changes proposed:
- update mean/var to set the proper split axis after the dimensional reduction
- mean/var now use factories.array instead of the DNDarray constructor

## Type of change

Select relevant options.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

Are all split configurations tested and accounted for?
[x] yes [ ] no
Does this change require a documentation update outside of the changes proposed?
[ ] yes [x] no
Does this change modify the behaviour of other functions?
[] yes [x] no
Are there code practices which require justification?
[] yes [x] no

### CODE REVIEW

File: heat/core/statistics.py [hunk 1]
```diff
@@ -760,88 +761,60 @@ def reduce_means_elementwise(output_shape_i):
...
+        else:
+            # singular axis given (axis) not equal to split direction (x.split)
+            return factories.array(
+                torch.mean(x._DNDarray__array, dim=axis),
+                is_split=x.split if axis > x.split else x.split - 1,
+            )
+    raise TypeError(
+        "axis (axis) must be an int or a list, ht.DNDarray, "
+        "torch.Tensor, or tuple, but was {}".format(type(axis))
+    )
```
[the remark this answers is not in the document]
Comment (coquelin77) [reply heat/core/statistics.py]: i find it likely that it could be a torch.Tensor, but it could be any iterable.

File: heat/core/statistics.py [hunk 2]
```diff
@@ -670,7 +670,8 @@ def mean(x, axis=None):
     Returns
     -------
     means : ht.DNDarray
-        The mean/s, if split, then split in the same direction as x.
+        The mean/s, if split, then split in the same direction as x, if possible. If axis > split
+        then the split is reduced by 1. If axis == split, then the resulting DNDarray is split None.
```
[the remark this answers is not in the document]
Comment (coquelin77) [reply heat/core/statistics.py]: you are correct, I reworded it with your suggestions.
